### PR TITLE
HDL refactoring

### PR DESCRIPTION
**Pull request by BFH-ktt1**

### CODE REVIEW

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/fpga/download/AlteraDownload.java [hunk 1]
```diff
@@ -114,8 +114,8 @@ public ProcessBuilder PerformStep(int stage) {
 
   @Override
   public boolean readyForDownload() {
-    final var SofFile = new File(SandboxPath + ToplevelHDLGeneratorFactory.FPGAToplevelName + ".sof").exists();
-    final var PofFile = new File(SandboxPath + ToplevelHDLGeneratorFactory.FPGAToplevelName + ".pof").exists();
+    final var SofFile = new File(SandboxPath + ToplevelHDLGeneratorFactory.FPGA_TOP_LEVEL_NAME + ".sof").exists();
```
[the remark this answers is not in the document]
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/download/AlteraDownload.java:117]: As this is already a big PR, I will concentrate on fpga/download/* later

File: src/main/java/com/cburch/logisim/fpga/download/AlteraDownload.java [hunk 2]
```diff
@@ -132,11 +132,11 @@ public ProcessBuilder DownloadToBoard() {
     command.add("jtag");
     command.add("-o");
     // if there is no .sof generated, try with the .pof
-    if (new File(SandboxPath + ToplevelHDLGeneratorFactory.FPGAToplevelName + ".sof").exists()) {
-      command.add("P;" + ToplevelHDLGeneratorFactory.FPGAToplevelName + ".sof"
+    if (new File(SandboxPath + ToplevelHDLGeneratorFactory.FPGA_TOP_LEVEL_NAME + ".sof").exists()) {
```
[the remark this answers is not in the document]
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/download/AlteraDownload.java:135]: As this is already a big PR, I will concentrate on fpga/download/* later

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java [hunk 3]
```diff
@@ -123,8 +123,8 @@ public static String GetZeroVector(int nrOfBits, boolean floatingPinTiedToGround
   public static String getConstantVector(long value, int nrOfBits) {
     final var bitString = new StringBuffer();
     var mask = 1L << (nrOfBits - 1);
-    if (HDL.isVHDL())
-      bitString.append('"');
+    if (HDL.isVHDL())
```
[the remark this answers is not in the document]
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java:130]: This is an old version of the file, there is a new push

[the remark this answers is not in the document]
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java:130]: Sorry, my bad, fixed

File: src/main/java/com/cburch/logisim/fpga/file/FileWriter.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import com.cburch.logisim.Main;
+import com.cburch.logisim.fpga.gui.Reporter;
+import com.cburch.logisim.fpga.hdlgenerator.HDL;
+import com.cburch.logisim.util.LineBuffer;
+
+public class FileWriter {
+
+  public static final String REMARK_LINE =
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/file/FileWriter.java:23]: Can we have this replaced with 

```java
public static final String REMARK_LINE = "-".repeat(81);
```

File: src/main/resources/resources/logisim/strings/fpga/fpga.properties [hunk 5]
```diff
@@ -2,6 +2,15 @@ FPGANotSupported = Not supported
 FPGASupported = Supported
 FPGAUnknown = Unknown
 #
+# file/FileWriter.java
+#
+fileCreateHDLFile = Creating HDL file : "%s"
```
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/fpga/fpga.properties:7]: IIRC space in front  `:` is not correct in English (multiple). Also may I suggest rearranging the strings, the the file name is always the last element, so no matter how long the path there shown is it won't degrade message readablity:

```
fileHDLFileExists = HDL file already exists: %s
fileCreateScriptFile = Creating script file: %s
fileScriptsFileExists = Script file already exists: %s
fileUnableToCreate = Unable to create file: %s
fileUnableToWrite = Could not write to file: %s
```

**Comment by BFH-ktt1** [comment]
Okay, This PR is already big enough. Ready for review and merge

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/fpga/file/FileWriter.java (hunk 4 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/file/FileWriter.java:23]: Done

File: src/main/resources/resources/logisim/strings/fpga/fpga.properties (hunk 5 above)
Comment (BFH-ktt1) [reply src/main/resources/resources/logisim/strings/fpga/fpga.properties]: done

**Review by MarcinOrlowski — APPROVED** [review]

**Pull request merged**